### DISCUSSION TITLE
how to change ui.input style?

### DISCUSSION THREAD

**Discussion by yanghw116** [zauberzeug/nicegui#2509] [Q&A] [answered]
### Question

hi everyone. I'm new to nicegui.

code example:
```py
from nicegui import ui

ui.input(label='kernel_shape', value='1, 1')
ui.input(label='pads', value='0, 0, 0, 0')
ui.input(label='strides', value='1, 1')

ui.run()
```

I create 3 `ui.input`. they look like:

![image](https://github.com/zauberzeug/nicegui/assets/156150671/eebf9704-c5f6-42c1-8432-c499e13730f5)

but I need they to be like (forget the `+` in the input box):

![image](https://github.com/zauberzeug/nicegui/assets/156150671/9354e7d0-ceab-41fb-a3bf-627b351c2ce9)

can anyone helps me? thanks in advance.

**Comment by yanghw116** [comment]
```
with ui.row().classes('w-full justify-right items-center'):
    ui.label('kernel_shape: ').classes('ml-auto')
    ui.input(value='1, 1').props('standout v-model="text" dense="dense" size=48')
with ui.row().classes('w-full justify-right items-center'):
    ui.label('pads: ').classes('ml-auto')
    ui.input(value='0, 0, 0, 0').props('standout v-model="text" dense="dense" size=48')
with ui.row().classes('w-full justify-right items-center'):
    ui.label('strides: ').classes('ml-auto')
    ui.input(value='1, 1').props('standout v-model="text" dense="dense" size=48')
```
the above code nearly solve this problem. it looks like
![image](https://github.com/zauberzeug/nicegui/assets/156150671/3d994938-fee9-4b56-8ff0-be7914da5a35)
then I found the space between rows is kind of large. How can I shrink the space?
@falkoschindler any idea?

**Reply by rodja** [reply]
try `gap-0` or `gap-1` Tailwind classes for each `ui.row()`.

**Reply by yanghw116** [reply]
```
with ui.row().classes('w-full justify-right items-center gap-0'):
    ui.label('kernel_shape: ').classes('ml-auto')
    ui.input(value='1, 1').props('standout v-model="text" dense="dense" size=48')
with ui.row().classes('w-full justify-right items-center gap-0'):
    ui.label('pads: ').classes('ml-auto')
    ui.input(value='0, 0, 0, 0').props('standout v-model="text" dense="dense" size=48')
with ui.row().classes('w-full justify-right items-center gap-0'):
    ui.label('strides: ').classes('ml-auto')
    ui.input(value='1, 1').props('standout v-model="text" dense="dense" size=48')
```
![image](https://github.com/zauberzeug/nicegui/assets/156150671/5aedfcf4-27ae-4bb6-91d5-c71438d0e3f1)
seems that the space between the 'ui.label' and 'ui.input' is shrinked, not that between the rows. 
Maybe it should be more specified. This space:
![image](https://github.com/zauberzeug/nicegui/assets/156150671/63771fb0-8bb0-4677-82cf-03203e750102)

**Reply by rodja** [reply, marked as answer]
Ah, sorry. I suggest your wrap your three rows in a `ui.column().classes('gap-0')`. Another approach would be to modify the global default gap with `ui.query('.nicegui-content').classes('gap-0')` similar to https://nicegui.io/documentation/query#modify_default_page_padding.

**Reply by yanghw116** [reply]
it's working. thank you very much.
